src/components/modebar/manage.js

```diff
@@ -0,0 +1,113 @@
+/**
+* Copyright 2012-2015, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Plotly = require('../../plotly');
+
+var createModebar = require('./');
+var modebarButtons = require('./buttons');
+
+/**
+ * Modebar wrapper around 'create' and 'update',
+ * chooses buttons to pass to Modebar constructor based on
+ * plot type and plot config.
+ *
+ * @param {object} gd main plot object
+ *
+ */
+module.exports = function manageModebar(gd) {
+    var fullLayout = gd._fullLayout,
+        context = gd._context,
+        modebar = fullLayout._modebar;
+
+    if(!context.displayModeBar && modebar) {
+        modebar.destroy();
+        delete fullLayout._modebar;
+        return;
+    }
+
+    if(!Array.isArray(context.modebarButtonsToRemove)) {
+        throw new Error([
+            '*modebarButtonsToRemove* configuration options',
+            'must be an array.'
+        ].join(' '));
+    }
+
+    var buttonGroups = getButtonGroups(fullLayout, context.modebarButtonsToRemove);
+
+    if(modebar) modebar.update(gd, buttonGroups);
+    else fullLayout._modebar = createModebar(gd, buttonGroups);
```

Comment:
N.B. 

A reference to the modebar instance is currently kept in `gd._fullLayout._modebar`.

So,

```js
var gd = document.getElementByIf('graph-div');
var modebar = gd._fullLayout._modebar;
modebar.update(gd, [[ { name: 'button 1', click: function() { console.log('hello world'); } } ]]);
```

update the modebar, display one button logging `'hello world'` on click.

Maybe we should place the reference to the modebar instance off `gd` directly (i.e. `gd._modebar`) to be easily to use.